mlflow/pyspark/optuna/study.py

```
@@ -0,0 +1,97 @@
+        n_jobs: int = 1,
+        catch: Iterable[type[Exception]] = (),
+        callbacks: Optional[Iterable[Callable[[Study, FrozenTrial], None]]] = None,
+    ) -> None:
+        experiment_id = self._storage._experiment_id
+        study_name = self._study_name
+
+        def run_task_on_executor_pd(iterator):
+            import optuna
+            import pandas as pd
+            import traceback
+
+            import mlflow
+            from mlflow.optuna.storage import MlflowStorage
+
+            mlflow.set_tracking_uri("databricks")
+
+
+            storage = MlflowStorage(experiment_id=experiment_id)
+            study = optuna.load_study(study_name=study_name, storage=storage)
+            for pdf in iterator:
+                try:
+                    error_message = []
+                    num_trials = len(pdf)
+                    _optimize_sequential(
+                        study,
+                        func,
+                        num_trials,
+                        timeout,
+                        catch,
+                        callbacks,
+                        False,
+                        False,
+                        datetime.datetime.now(),
+                        None,
+                    )
+                    error_message.append(None)
+                except BaseException as e:
+                    _traceback_string = traceback.format_exc()
+                    e._tb_str = _traceback_string
+                    error_message.append(_traceback_string)
+                finally:
+                    df = pd.DataFrame(
+                        {
+                            "error": error_message,
+                        }
+                    )
+                    yield df
+
+        num_tasks = n_trials
+        if n_jobs == -1:
+            n_jobs = num_tasks
+        input_df = self.spark.range(start=0, end=num_tasks, step=1, numPartitions=n_jobs)
+
+        return input_df.mapInPandas(
```

To keep consistent with original optuna `optimize` API,
when exception occurs, `optimize` should raise the exception,

but the code returns a list of rows that contains the error message.